### PR TITLE
related to issue #748, solution result from #issuecomment-331347264; …

### PR DESCRIPTION
**Pull request by japrogramer**
…test client encodes headers correctly.

### CODE REVIEW

**Comment by japrogramer** [comment]
@andrewgodwin 
"The ASGI design decision is to transport both request and response headers as lists of 2-element [name, value] lists and preserve headers exactly as they were provided."
Ok, well the tests that are failing aren't treading the headers like of list of 2 element lists. They are treating 
them as a dict.

File: channels/test/websocket.py [hunk 1]
```diff
@@ -82,6 +82,9 @@ def send(self, to, content={}, text=None, path='/'):
         self.channel_layer.send(to, self._get_content(content, text, path))
         self._session_cookie = False
 
+    def _tuple_headers(self):
```
Comment (proofit404) [inline channels/test/websocket.py]: Why you call it `_tuple_headers`? It returns a list of lists.

File: channels/test/websocket.py [hunk 2]
```diff
@@ -82,6 +82,9 @@ def send(self, to, content={}, text=None, path='/'):
...
+        return [[x, self.headers[x]] for x in self.headers]
```
Comment (proofit404) [inline channels/test/websocket.py]: Maybe `[[header, value] for header, value in self.headers.items()]` is more readable.

Comment (japrogramer) [reply channels/test/websocket.py]: Ill make those changes, but than the tests would still fail.

Comment (proofit404) [reply channels/test/websocket.py]: You need to edit tests as well ;-)

File: tests/test_wsclient.py [hunk 3]
```diff
@@ -52,8 +52,8 @@ def test_session_in_headers(self):
         self.assertEqual(content['path'], '/')
 
         self.assertTrue('headers' in content)
-        self.assertTrue('cookie' in content['headers'])
-        self.assertTrue(b'sessionid' in content['headers']['cookie'])
+        self.assertTrue([False, True]['cookie' in [x[0] for x in content['headers']]])
```
Comment (proofit404) [inline tests/test_wsclient.py]: You are using freaky item access. Rewrite this expression in a readable way. Better to use `assertIn` method here.

File: tests/test_wsclient.py [hunk 4]
```diff
@@ -53,7 +53,7 @@ def test_session_in_headers(self):
 
         self.assertTrue('headers' in content)
         self.assertIn('cookie', [x[0] for x in content['headers']])
-        self.assertIn(b'sessionid', [x[1] for x in content['headers'] if x[0] is 'cookie'])
```
Comment (proofit404) [inline tests/test_wsclient.py]: Please provide me with content value and I'll try to improve this.

Comment (japrogramer) [reply tests/test_wsclient.py]: this is what i got.

{'reply_channel': 'defaultqFyJJ', 'path': '/', 'headers': [['cookie', b' sessionid=chn75dd650d27ba86a8f0f0617567dd9']]}

**Comment by japrogramer** [comment]
@proofit404 I made the changes request and now all tests pass

**Comment by andrewgodwin** [comment]
Waiting for @proofit404 to approve and we can merge (I'm OK with this).

**Comment by proofit404** [comment]
Hi, guys!

Sorry I wasn't responsive last weeks. I resolve all issues in my personal life and now fully available as channels maintainer.

**Comment by andrewgodwin** [comment]
No problem, I would have merged it anyway after a few days :)

**Pull request merged**